admin_pages/events/Events_Admin_Page.core.php

```diff
@@ -2131,58 +2134,165 @@ protected function generateDeletionPreview($event_ids)
         );
     }
 
+    /**
+     * Checks for a POST submission
+     * @since $VID:$
+     */
+    protected function confirmDeletion()
+    {
+        $deletion_job_code = isset($this->_req_data['deletion_job_code']) ? $this->_req_data['deletion_job_code'] : '';
+        $models_and_ids_to_delete = $this->getModelsAndIdsToDelete($deletion_job_code);
+        $form = new ConfirmEventDeletionForm($models_and_ids_to_delete['Event']);
+        // Initialize the form from the request, and check if its valid.
+        $form->receive_form_submission($this->_req_data);
+        if ($form->is_valid()) {
+            // Redirect the user to the deletion batch job.
+            EEH_URL::safeRedirectAndExit(
+                EE_Admin_Page::add_query_args_and_nonce(
+                    array(
+                        'page'        => 'espresso_batch',
+                        'batch'       => EED_Batch::batch_job,
+                        'deletion_job_code' => $deletion_job_code,
+                        'job_handler' => urlencode('EventEspressoBatchRequest\JobHandlers\ExecuteBatchDeletion'),
+                        'return_url'  => urlencode(
+                            add_query_arg(
+                                [
+                                    'status' => 'trash'
+                                ],
+                                EVENTS_ADMIN_URL
+                            )
+                        )
+                    ),
+                    admin_url()
+                )
+            );
+        } else {
```

Comment:
i think this entire else block (not the code within it, but just the actual `else` structure)
might be unnecessary because `EEH_URL::safeRedirectAndExit()` calls `exit()` after the redirect,
so the else block will not ever get called if the form is valid.

It is essentially the same as:

```
if ( valid ) {
	// do stuff
	exit();
}
// don't need an else here because of the above exit
// if the form is not valid, it won't redirect and exit,
// and will end up here anyways
```

Reply:
oh thanks! yes